app/digitizingcontroller.cpp

```diff
@@ -200,11 +244,7 @@ void DigitizingController::onPositionChanged()
     return;
 
   QgsPoint point = mPositionKit->position();
-  QgsGeometry geom( point.clone() );
-  geom.transform( transformer() );
-
-  QgsPoint *layerPoint = qgsgeometry_cast<QgsPoint *>( geom.get() );
-  fixZ( layerPoint );
+  QgsPoint *layerPoint = getLayerPoint( point, true );
```

Comment:
layerPoint here is not getting deleted... maybe better to return std::unique_ptr<QgsPoint> from getLayerPoint() to make things safer ?